@xen-orchestra/proxy/src/app/mixins/backups/index.js

```diff
@@ -56,7 +56,16 @@ export default class Backups {
     let run = ({ xapis, ...rest }) =>
       new Backup({
         ...rest,
-        app,
+
+        // don't change config during backup execution
+        config: app.config.get('backups'),
+        getAdapter: Disposable.factory(function* (remoteId) {
+          const adapter = yield app.remotes.getAdapter(this._remotes[remoteId])
+          return {
+            adapter,
+            remoteId,
```

Comment:
Wouldn't it be simpler to pass `app.remotes.getAdapter.bind(app.remotes)` and make adapters exposing a `remoteId` property?

Reply:
This is only needed for backups, I don"t think is necessary to do it generally.

Reply:
The adapters are only meant for backups.

And I feel like this function is too much linked to the internal implementation of `Backup`, calling code should simply provide access to the remote handler/adapter.

Reply:
Adapters are also used for file restoration. To not misunderstood you. You want this [method](https://github.com/vatesfr/xen-orchestra/blob/ff4b51d276a2fd265e8319073ef11df553a24b4e/%40xen-orchestra/proxy/src/app/mixins/remotes.js#L64-L69) to return `{ adapter, remoteId }`?

Reply:
No, I want either:
-  `RemoteAdapter` to expose a `remoteId` prop ;
- or `Backup` to construct `{ adapter, remoteId }` itself.

This structure is not natural in itself and the calling code should not have to take care of it.

Reply:
Also, remote adapters are dedicated to list, read, write and delete backups, they probably won't be used for anything not related to the backup, therefore `Remotes#getAdapter` should be moved into backup-specific code.

Reply:
`RemoteAdapter` cannot expose the `remoteId` because it's an app problematic, a remote is just an URL and options that allow it to connect and do some operations.

Reply:
To resume you want to:
1. move `Remotes#getAdapter` to `Backups`, I can do it in an other PR because it can lead to unrelated changes
2. passe `getAdapter` to `Backup` which will override it